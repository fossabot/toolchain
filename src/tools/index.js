export { default as logger } from './logger';
export { default as metrics } from './metrics';
export { default as tracer } from './tracer';
