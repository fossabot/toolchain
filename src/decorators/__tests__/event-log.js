import { logger } from 'tools';

import eventLog from '../event-log';

logger.info = jest.fn();
logger.error = jest.fn();

describe('eventLog', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  describe('when logger not present in context', () => {
    it('skip the log and return the result in await', async () => {
      const original = () => 'yes';
      const decorated = eventLog()(original);
      const result = await decorated();

      expect(result).toBe('yes');
      expect(logger.info.mock.calls.length).toBe(0);
    });

    it('throws error from the original function', async () => {
      const original = () => {
        const e = { message: 'error' };
        throw e;
      };
      const decorated = eventLog()(original);

      try {
        await decorated();
      } catch (e) {
        expect(e.message).toBe('error');
      }
    });
  });

  describe('when logger present in context', () => {
    it('log the event when input function call succeeds', async () => {
      const original = async () => 'yes';
      const decorated = eventLog()(original);
      const result = await decorated({}, {}, { logger });

      expect(result).toBe('yes');
      expect(logger.info.mock.calls).toMatchSnapshot();
    });

    it('log the error event when input function fails', async () => {
      const original = () => {
        const e = { message: 'error' };
        throw e;
      };
      const decorated = eventLog()(original);

      try {
        await decorated({}, {}, { logger });
      } catch (e) {
        expect(e.message).toBe('error');
        expect(logger.info.mock.calls).toMatchSnapshot();
        expect(logger.error.mock.calls).toMatchSnapshot();
      }
    });
  });
});
