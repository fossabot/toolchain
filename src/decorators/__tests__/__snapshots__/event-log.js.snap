// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`eventLog when logger present in context log the error event when input function fails 1`] = `Array []`;

exports[`eventLog when logger present in context log the error event when input function fails 2`] = `
Array [
  Array [
    Object {
      "error": Object {
        "message": "error",
      },
      "event": "original",
    },
    "error",
  ],
]
`;

exports[`eventLog when logger present in context log the event when input function call succeeds 1`] = `
Array [
  Array [
    Object {
      "event": "original",
    },
    "original",
  ],
]
`;
